robottelo/hosts.py

```diff
@@ -1392,15 +1394,21 @@ def is_upstream(self):
         :return: True if no downstream satellite RPMS are installed
         :rtype: bool
         """
-        return self.execute('rpm -q satellite-capsule &>/dev/null').status != 0
+        return self.execute(f'rpm -q {self.product_rpm_name} &>/dev/null').status != 0
+
+    @cached_property
+    def is_stream(self):
+        """Check if the Capsule is a stream release or not
+
+        :return: True if the Capsule is a stream release
+        :rtype: bool
+        """
+        return 'stream' in self.execute(f'rpm -q {self.product_rpm_name}').stdout.strip()
 
     @cached_property
     def version(self):
-        if not self.is_upstream:
-            version = self.execute('rpm -q satellite-capsule').stdout
-            return 'stream' if 'stream' in version else version.split('-')[2]
-        else:
-            return 'upstream'
+        rpm_name = self.upstream_rpm_name if self.is_upstream else self.product_rpm_name
+        return self.execute(f'rpm -q {rpm_name}').stdout.split('-')[2]
```

Comment:
```
In [1]: test = 'satellite-6.13.0-6.el8sat.noarch'

In [2]: test.split('-')[2]
Out[2]: '6.el8sat.noarch'

In [7]: test.split('-')[1]
Out[7]: '6.13.0'
```

Reply:
as this will be inherited to the Satellite class, this will not work for the Satellite class

Reply:
Note for myself: use `rpm --qf %{version}`

Reply:
you mean `rpm -q --qf '%{VERSION}' satellite`

Reply:
Fixed. Thanks, Omkar for spotting it.